ci.md

```diff
@@ -0,0 +1,201 @@
+               `Test Result`. You can also check the `Console Output` for the
+               full test logs.
+    * Actions: When there are test failures, the build is marked red ❌.
+               On the build logs, the test failures should be under the
+               `make test` collapsible.
+    * Check which tests are known failures by searching for the test name on
+      issues.
+* There are warnings. Warnings that aren't pre-existing must be fixed.
+    * Jenkins: When there are warnings, the build is marked yellow 🟡.
+               On the build's page, the warnings are listed under
+               `GNU C Compiler` and `CMake`. You can also check the
+               `Console Output` for the full warning logs.
+    * Actions: Does not detect compiler or CMake warnings.
+* There are static checker failures. All code checker failures must be fixed.
+    * Jenkins: Does not run static checkers.
+    * Actions: When there are code cherk failures, the build is marked red ❌.
+               On the build logs, the checker failures should be under the
+               `Code check` collapsible.
+* There are infrastructure failures. These should be reported to the build farmer.
+    * Jenkins: When there's an infra failure, the build is marked red 🔴.
+               These can manifest in various ways. In general, if the
+               build logs don't fall in the categories above, there's a high
+               chance it's an infrastructure failure. Common patterns usually
+               have several `java` or `hudson` messages.
+    * Actions: Infrastructure failures haven't been identified.
+
+## Triggering CI
+
+CI is usually triggered automatically when code is pushed or a pull request is
+opened. But often it's necessary to re-trigger CI without making changes to the
+code if:
+
+* There were infrastructure failures in a previous build.
+* There are fixes upstream, on infra or dependencies, that may make a new build pass.
+
+The following methods can be used to re-trigger builds:
+
+* Jenkins
+    * Make a comment starting with `@osrf-build run tests` on the pull request
+      and all Jenkins builds will be re-triggered.
+    * To restart just one specific build and avoid re-running builds that aren't
+      necessary, go to the failing build and click `Retry`. This is only available
+      to maintainers.
+* Actions
+    * On the top-right of a build, click `Re-run jobs`. This button is only
+      available to maintainers and sometimes mysteriously disappears.
+
+### Custom branches
+
+Maintainers can manually trigger builds for custom branches without opening pull
+requests, optionally building dependencies from source and changing other
+configurations for testing.
+
+* Actions: Builds are run for every commit. For building dependencies from source,
+           add a `.github/ci/dependencies.yaml` file as described
+           [here](https://github.com/ignition-tooling/action-ignition-ci#source-dependencies).
```

Comment:
I think I'm just not understanding this because I've never had to do this. Is this for if you are adding things that will require new dependencies, and this adds dependencies to CI so that you can test that it works, before officially adding a dependency to the repo?
Is there an example for this bullet? I think the example in the next bullet is helpful (though I've never had to do that either, so can't comment on that).

Reply:
I'll add more information about when this may be needed. It's tough to be concise and thorough at the same time :sweat_smile: 

One possible reason to use this is testing changes across repos. If your `ign-gazebo` PR needs a change from `ign-physics`, you could build `ign-physics` from source to check it works on CI before even opening the `ign-physics` PR, or making a release.

Reply:
Ah okay. Then I've actually come across the need for that but wasn't aware how it can be done. I think saying that scenario helps.

Reply:
On a related note, I just looked at this link, and I think it'd be nice to clarify what to do for `Be sure to add the apt dependencies of dependencies build from source to apt-dependencies`, because I honestly have no clue how to do this after reading that sentence :upside_down_face:

Reply:
If we later want to open a PR, would we have to remove the dependency file?

Reply:
Ok, this _should_ make things clearer: https://github.com/ignition-tooling/action-ignition-ci/pull/37